server/src/controllers/Transaction.controller.js

```diff
@@ -73,7 +73,85 @@ export default class TransactionController {
     return res.status(200).json({
       status: 200,
       data,
-      message: 'Transaction was successful'
+      message: `N${newTransaction.amount} was credited to your account`
+    });
+  }
+
+  /**
+   * @method debitTransaction
+   *
+   * @param {object} req
+   * @param {object} res
+   *
+   * @returns {object} status code, data and message properties
+   */
+  static debitTransaction(req, res) {
+    const { accountNumber } = req.params;
+    const { amount } = req.body;
+    const { role, id } = req.decoded;
+    const minimumBalance = 500;
+
+    if (role !== 'staff') {
+      return res.status(401).json({
+        status: 401,
+        error: 'You are not authorized to carry out that action'
+      });
+    }
+
+    const accountToDebit = accounts
+      .find(account => account.accountNumber === parseInt(accountNumber, 10));
+
+    if (!accountToDebit) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Account does not exist'
+      });
+    }
+
+    const { accountNumber: userAccountNumber, balance } = accountToDebit;
+    const balanceWithdrawable = balance - minimumBalance;
+
+    if (amount > balanceWithdrawable) {
+      return res.status(400).json({
+        status: 400,
+        error: 'Insufficient funds, cannot perform transaction'
+      });
+    }
```

Comment:
Good job, please also include that inactive accounts can not be debited from. Add the appropriate tests too